sw/device/rom_exts/rom_ext.c

```diff
@@ -0,0 +1,48 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/rom_exts/rom_ext.h"
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/stdasm.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/print.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+
+static dif_uart_t uart;
+
+void rom_ext_boot(void) {
+  dif_uart_result_t init_result = dif_uart_init(
+      (dif_uart_params_t){
+          .base_addr = mmio_region_from_addr(TOP_EARLGREY_UART_BASE_ADDR),
+      },
+      &uart);
+
+  if (init_result != kDifUartOk) {
+    abort();
+  }
+
+  dif_uart_config_result_t config_result =
+      dif_uart_configure(&uart, (dif_uart_config_t){
+                                    .baudrate = kUartBaudrate,
+                                    .clk_freq_hz = kClockFreqPeripheralHz,
+                                    .parity_enable = kDifUartToggleDisabled,
+                                    .parity = kDifUartParityEven,
+                                });
+
+  if (config_result != kDifUartConfigOk) {
+    abort();
+  }
```

Comment:
Please can you add a TODO to decide what state the ROM_EXT receives things like the UART in - I'm not convinced it should need reconfiguring here - but this is something we need to decide as part of the handoff between Mask ROM and ROM_EXT.

Reply:
I will drop a `TODO`, I also start an issue to accumulate these questions, otherwise they often get lost.

Reply:
Done.